lib/nerves_hub_cli.ex

```diff
@@ -20,10 +22,10 @@ defmodule NervesHubCLI do
     override_dir =
       Application.get_env(:nerves_hub_cli, :home_dir) || System.get_env("NERVES_HUB_HOME")
 
-    if override_dir == nil or override_dir == "" do
-      Path.expand("~/.nerves-hub")
-    else
-      override_dir
+    case {override_dir, System.get_env("XDG_DATA_HOME")} do
+      {dir, _} when valid_home_dir(dir) -> dir
+      {_, xdg} when valid_home_dir(xdg) -> Path.join(xdg, "nerves-hub")
+      _ -> Path.expand("~/.nerves-hub")
     end
```

Comment:
This function feels like it needs some TLC. Maybe something along the lines of this:

```elixir
defp valid_home_dir(dir) do
  is_binary(dir) and dir != ""
end

def home_dir() do
  [[Application.get_env(:nerves_hub_cli, :home_dir)],
   [System.get_env("NERVES_HUB_HOME")],
   [System.get_env("XDG_DATA_HOME"), "nerves-hub"],
   ["~", ".nerves-hub"]]
  |> Enum.find(fn [dir | _] -> valid_home_dir(dir) end)
  |> Path.join()
  |> Path.expand()
end
```

One accidental side effect of this is that all of the options get Path expanded and that seems desirable so that you can set overrides with `~` characters in the paths.

Reply:
I like it. Updated